feat: automatically set dns service address

prometherion opened the pull request:
Closes #468, pinging the main stakeholder such as @kvaps and the IPv6 gurus @johannwagner @hexchen

When defining a different Service CIDR and an empty DNS Service one, Kamaji will automatically compute the correct value.

Once applying this Tenant Control Plane snippet:

```yaml
  networkProfile:
    serviceCidr: 10.20.0.0/16
    port: 6443
```

This is the resulting one stored in the cluster.

```yaml
  networkProfile:
    clusterDomain: cluster.local
    dnsServiceIPs:
    - 10.20.0.10
    podCidr: 10.244.0.0/16
    port: 6443
    serviceCidr: 10.20.0.0/16
```

Tested also with IPv6:

```yaml
  networkProfile:
    serviceCidr: 2002::1234:abcd:ffff:c0a8:101/64
    port: 6443
```

Defaulted values:

```yaml
    version: v1.30.0
  networkProfile:
    clusterDomain: cluster.local
    dnsServiceIPs:
    - 2002::1234:abcd:ffff:c0a8:111
    podCidr: 10.244.0.0/16
    port: 6443
    serviceCidr: 2002::1234:abcd:ffff:c0a8:101/64
```

prometherion committed:
feat: automatically set dns service address

Signed-off-by: Dario Tranchitella <dario@tranchitella.eu>

prometherion committed:
feat(helm): automatically set dns service address

Signed-off-by: Dario Tranchitella <dario@tranchitella.eu>

prometherion committed:
docs: automatically set dns service address

Signed-off-by: Dario Tranchitella <dario@tranchitella.eu>

johannwagner commented:
Hey, thanks a lot for this fix. 
The IPv6 given service CIDR look wrong, I would expect something like  2001:db8:abcd:efgh::/64 as CIDR with 2001:db8:abcd:efgh::10 as DNS IP.

prometherion commented:
Are you sure the `2001:db8:abcd:efgh::` address is correct? I'm getting a [network validation error](https://www.vultr.com/resources/subnet-calculator-ipv6/?prefix_length=64&display=long&ipv6_address=2001%3Adb8%3Aabcd%3Aefgh%3A%3A%2F64).

~Also, I'm a bit confused: why `10` instead of `a`? If I understood correctly, the octet is way bigger in IPv6, so `010` in IPv4 would be translated into `000a` in IPv6.~

Ok, maybe I'm getting your point: in IPv4 we can expect a +10 bit in the last octet, and +16 in IPv6?

johannwagner commented:
Whoopsie, I fucked up the example. 2001:db8:abcd::/64 is correct. The other one also has host bits set, which is not correct. 

I guess so, they specify :10 as DNS address, which is 16 bits added to the network address.

prometherion commented:
Thanks, I added the required changes, such as:
- adding 10 bit for IPv4
- adding 16 bits for IPv6

I updated the initial comment on the issue just for reference.
